sdk/core/Azure.Core/tests/TestFramework/TestRecording.cs

```diff
@@ -0,0 +1,156 @@
+    {
+        private const string RandomSeedVariableKey = "RandomSeed";
+
+        public TestRecording(RecordedTestMode mode, string sessionFile, RecordedTestSanitizer sanitizer, RecordMatcher matcher)
+        {
+            Mode = mode;
+            _sessionFile = sessionFile;
+            _sanitizer = sanitizer;
+            _matcher = matcher;
+        }
+
+        public RecordedTestMode Mode { get; }
+
+        private readonly string _sessionFile;
+
+        private readonly RecordedTestSanitizer _sanitizer;
+
+        private readonly RecordMatcher _matcher;
+
+        private RecordSession _session;
+
+        private Random _random;
+
+        public Random Random
+        {
+            get
+            {
+                if (_random == null)
+                {
+                    switch (Mode)
+                    {
+                        case RecordedTestMode.None:
+                            _random = new Random();
+                            break;
+                        case RecordedTestMode.Record:
+                            _random = new Random();
+                            int seed = _random.Next();
+                            _session.Variables[RandomSeedVariableKey] = seed.ToString();
+                            _random = new Random(seed);
+                            break;
+                        case RecordedTestMode.Playback:
+                            _random = new Random(int.Parse(_session.Variables[RandomSeedVariableKey]));
+                            break;
+                        default:
+                            throw new ArgumentOutOfRangeException();
+                    }
+                }
+                return _random;
+            }
+        }
+
+        private RecordSession Load()
+        {
+            using FileStream fileStream = File.OpenRead(_sessionFile);
+            using JsonDocument jsonDocument = JsonDocument.Parse(fileStream);
+            return RecordSession.Deserialize(jsonDocument.RootElement);
+        }
+
+        public void Start()
+        {
+            switch (Mode)
+            {
+                case RecordedTestMode.Record:
+                    _session = new RecordSession();
+                    break;
+                case RecordedTestMode.Playback:
+                    _session = Load();
+                    break;
+            }
+        }
+
+        public void Stop()
+        {
+            if (Mode == RecordedTestMode.Record)
+            {
+                var directory = Path.GetDirectoryName(_sessionFile);
+                Directory.CreateDirectory(directory);
+
+                using FileStream fileStream = File.Create(_sessionFile);
+                var utf8JsonWriter = new Utf8JsonWriter(fileStream, new JsonWriterOptions()
+                {
+                    Indented = true
+                });
+                _session.Sanitize(_sanitizer);
+                _session.Serialize(utf8JsonWriter);
+                utf8JsonWriter.Flush();
```

Comment:
I wouldn't block on this now, but it'd be awesome if I could diff two RecordSessions relative to the headers we actually compare.  I imagine most of the time I run the live tests to update, it's going to result in a very noisy diff to see if anything meaningful changed.

Reply:
Do you mean programmatically or with something like a diff tool?

Reply:
I was specifically thinking about something programmatic to only copy sessions that have "meaningful" changes, for some definition of meaningful.  Right now the workflow is I'll make a change to one API, run the tests with recordings on, and get a session directory full of files.  I copy and paste that session directory over the old one and my diffs look like a red/green explosion with all the dates/account names/etc changing for every call -- even though the meaningful changes are restricted to one API.

Another way of solving this would be to dump out something like `[Ignored]` for headers RecordMatcher doesn't consider.

Reply:
I see.

I don't want to add too many smarts right now because I only have AppConfig as an example and things that work for it might not work for other SDKs. I will pay attention to this, making things easy and less noisy is my goal.